### ISSUE TITLE
Five-server is not open source

### ISSUE THREAD

**Issue by OIRNOIR** [yandeu/five-server#148]
Rather than using an open source license, such as the MIT license (which is the license used by the now-dead [live-server package](https://github.com/tapio/live-server), five server uses a proprietary license that restricts any kind of modification, redistribution, or use. In other words, that makes this software "source-available"—except we would be violating the terms of the license to even examine the code in this repository according to restriction B of the Five Server License.

I don't know how much of this was intended, but I would recommend switching to an open source license like the GPL or MIT licenses. To me, it seems like this project intends to be open source, but hasn't been licensed as such yet.

**Comment by OIRNOIR** [comment]
Upon a closer look, your three-year-old version of [DiffDOM](https://github.com/fiduswriter/diffDOM) (here located at /public/scripts/) was released under the GNU Lesser General Public License v3.0, which requires all modifications and redistributions of such licensed work to be licensed under the same license or under another GPLv3 compatible license. "Copyright and license notices must be preserved." You may wish to correct this.

One way this can be fixed is by licensing the whole project as GPLv3. However, you may not wish to do this.
Another way to fix this is to license most of the project as MIT or some other license, but to add a note that some parts (including DiffDOM) are licensed under the GNU Lesser GPLv3.

**Comment by yandeu** [comment]
My intend was a private matter. I tried to licence the project as a whole very strict, but not the individual files. Some are still MIT. I guess I have missed the license of DiffDOM. Not sure why. Is the file in five-server really from the repo you mentioned?

---

> even examine the code in this repository according to restriction B of the Five Server License.

How so?

> To me, it seems like this project intends to be open source, but hasn't been licensed as such yet.

Kind of, yes. I have some issues with the license I haven't figured out how to solve yet.

**Comment by OIRNOIR** [comment]
Sorry for the giant essay.

I understand your desire to license it strictly, and you can definitely do it that way if you wish, but I don't think I want to use or contribute to such a project, as a matter of personal conviction, and I think many other people would agree with me. Namely, I don't want to violate the license, but using the software itself in any way without express permission from you violates it. Explanation further down.

> Is the file in five-server really from the repo you mentioned?

Yes, you can check their commit history and verify that the file in their repo looks to be the same as the one in this repo at the time it was committed.

> How so?

One of the restrictions in the license is that we are not allowed to "Decompile, disassemble or reverse engineer the Software or attempt to do any such things." If the source code were public (as it is), wouldn't it be considered reverse engineering to simply browse the code? The wording in this section is a bit strange to me, because this program is not compiled and its code is meant to be open source.

I also just wanted to analyze the issues I personally see with the current license, preventing it from being open source:
All four restrictions listed in the license go against the GNU's definition of free software. [They list four freedoms](https://www.gnu.org/licenses/quick-guide-gplv3.html):
1. the freedom to use the software for any purpose,
2. the freedom to change the software to suit your needs,
3. the freedom to share the software with your friends and neighbors, and
4. the freedom to share the changes you make.

Note that these freedoms must be upheld for all software licensed by an open source license. If you do not wish to license this project as open source, that's your decision to make.

Restriction (a) of the Five Server License removes freedoms (2) and (4) because it disallows modifying the software for your own needs, locally on your own computer. This includes making contributions such as bug fixes: If someone wishes to submit a pull request to fix a problem, they would have to make modifications to the code on their device before submitting their modifications as a pull request. This clause currently prohibits other people from contributing code to the project, as well as prohibits people from making derivative works. If this was your intention, I respect your decision, but I wanted to make sure you were aware.

Restriciton (b) of the Five Server License removes all of these freedoms because it prohibits inspecting, reviewing, and understanding the code itself.

Restriction (c) of the Five Server License removes freedoms (3), and (4) because it prohibits sharing the code without permission, **but** it also removes freedom (1) due to its "or otherwise use" clause. This clause, essentially, means that no one is free to use the software at all without your express permission. This would make sense if this were a closed source project to be used within a company or just by yourself and your friends, but it doesn't make much sense to me because it is currently published on NPM and as a VSCode extension. Wouldn't this clause mean that everyone using this program is automatically violating the license? I'm sure this is unintentional.

Restriction (d) of the Five Server License seems to make no sense. Does Five Server send notifications? Would it violate the license prohibit me from dismissing a notification that appeared on my device, since it prohibits hiding notifications? I would need more information to properly analyze that.

In accordance with the current license, I have uninstalled Five Server from my device, because to otherwise use the software would be in violation of section 2(c) of the license.

I support your right to license your software however you want, but the current license means that every user of Five Server is currently violating section 2(c) of the license, merely by using the software. In addition, making code contributions is also disallowed via section 2(a). Section 2(b) prohibits viewing the code, at which point it makes no sense to place the code on github, and section 2(d) seems to make no sense to me. If you could update your license to more effectively describe what we can and cannot do with your code, I would greatly appreciate it. I think it would be a great contribution to society if you made your code open source, but this is your decision to make.

**Comment by OIRNOIR** [comment]
Tl;DR: We agree that there are a number of issues with the current license. I respect your wishes to license your software however you want, but the current license's issues prevent me from using the software. I would prefer if you relicensed the project as an open source project, but your license is up to you.

**Comment by yandeu** [comment]
I have worked on a modification. What do you think?

```
2. RESTRICTIONS
You are NOT permitted to:
  a)  Share, distribute or resell the whole or any part of the Software for
      any commercial or non-commercial purpose except where permitted by
      item 3 Private Use and item 4 Contribute.

3. PRIVATE USE
You have the freedom to:
  a)  Use the Software for any purpose.
  b)  Change the Software to suit your needs.
  c)  Share the Software and changes you make within a circle of persons
      closely connected to you, such as relatives or friends.

4. CONTRIBUTE
You are permitted to:
  a)  Create a fork of the repository with the intention to committing back
      your changes to the upstream repository.
```

**Comment by OIRNOIR** [comment]
> I have worked on a modification. What do you think?

It still isn't open source, but I recognize and respect that that isn't your intention. For your purposes, it seems like it suits Five Server much better than the current license.

In other words, LGTM
